portfolio/src/main/webapp/script.js

```diff
@@ -63,4 +63,59 @@ function random_image() {
   // Remove the previous image.
   imageContainer.innerHTML = '';
   imageContainer.appendChild(imgElement);
+}
+
+
+//================================================================
+
+
+/**
+ * Javascript Fetches the content from the server and adds it to the DOM.
+ */
+function getHelloRosanna() {
+  console.log('Fetching the content from the serlvet.');
+
+  // The fetch() function returns a Promise because the request is asynchronous.
+  /* fetch('/data') sends a request to the /data URL. 
+  The server responds to this request with the content , 
+  exactly like when you navigated to the URL with your browser.
+  
+  The fetch() function makes a request in the background, 
+  so it returns a Promise, which is stored in the responsePromise variable
+
+  #MOST IMPORTANT: SENDS REQUEST TO /DATA*/
+  const responsePromise = fetch('/data');
+
+  /*
+    the responsePromise variable points to a Promise, and the code call the then() func, 
+    passing the handleresponse function in as an argument.
+    JS can reference a function using the name.
+  */
+  // When the request is complete, pass the response into handleResponse().
+  // tells JavaScript to call the handleResponse() function when the server returns a response.
+  responsePromise.then(handleResponse);
```

Comment:
What happens here if the fetch() results in an error?

Reply:
should i have a catch for error catching?

Reply:
In general, yes, you should ensure you're doing something (at least logging) error conditions. It helps a lot when troubleshooting later if an unexpected error starts happening.